docs/standard/exceptions.md

```diff
@@ -0,0 +1,528 @@
+| [NullReferenceException](https://docs.microsoft.com/dotnet/core/api/System.NullReferenceException) | [Exception](https://docs.microsoft.com/dotnet/core/api/System.Exception) | Thrown by the runtime only when a null object is referenced. | `object o = null; o.ToString();` |
+| [InvalidOperationException](https://docs.microsoft.com/dotnet/core/api/System.InvalidOperationException) | [Exception](https://docs.microsoft.com/dotnet/core/api/System.Exception) | Thrown by methods when in an invalid state. | Calling `Enumerator.GetNext()` after removing an Item from the underlying collection. |
+| [ArgumentException](https://docs.microsoft.com/dotnet/core/api/System.ArgumentException) | [Exception](https://docs.microsoft.com/dotnet/core/api/System.Exception) | Base class for all argument exceptions. | None (use a derived class of this exception). |
+| [ArgumentNullException](https://docs.microsoft.com/dotnet/core/api/System.ArgumentNullException) | [Exception](https://docs.microsoft.com/dotnet/core/api/System.Exception) | Thrown by methods that do not allow an argument to be null. | `String s = null; "Calculate".IndexOf (s);` |
+| [ArgumentOutOfRangeException](https://docs.microsoft.com/dotnet/core/api/System.ArgumentOutOfRangeException) | [Exception](https://docs.microsoft.com/dotnet/core/api/System.Exception) | Thrown by methods that verify that arguments are in a given range. | `String s = "string"; s.Chars[9];` |
+
+## How to use the try/catch block to catch exceptions
+
+Place the sections of code that might throw exceptions in a **try** block and place code that handles exceptions in a **catch** block. The **catch** block is a series of statements beginning with the keyword **catch**, followed by an exception type and an action to be taken.
+
+The following code example uses a **try**/**catch** block to catch a possible exception. The `Main` method contains a **try** block with a [StreamReader](https://docs.microsoft.com/dotnet/core/api/System.IO.StreamReader) statement that opens a data file called `data.txt` and writes a string from the file. Following the **try** block is a **catch** block that catches any exception that results from the **try** block.
+
+C#
+```
+using System;
+using System.IO;
+
+public class ProcessFile
+{
+    public static void Main()
+    {
+        try
+        {
+            StreamReader sr = File.OpenText("data.txt");
+            Console.WriteLine("The first line of this file is {0}", sr.ReadLine());
+	        sr.Dispose();
+        }
+        catch (Exception e)
+        {
+            Console.WriteLine("An error occurred: '{0}'", e);
+        }
+    }
+}
+```
+
+The common language runtime catches exceptions that are not caught by a catch block. Depending on how the runtime is configured, a debug dialog box appears, or the program stops executing and a dialog box with exception information appears, or an error is printed out to STDERR.
```

Comment:
@tdykstra I don't know. I also don't know if it would be the same if you build a native app. From the application point of view, it may not matter (so using a general term might be wise). @terrajobst or someone on his time might have the best insight.